FinSearchUnified/Tests/BusinessLogic/Models/FindologicArticleModelTest.php

```diff
@@ -0,0 +1,138 @@
+                    'name' => 'FindologicArticle 1',
+                    'active' => true,
+                    'tax' => 19,
+                    'supplier' => 'Findologic',
+                    'categories' => [
+                        ['id' => 3],
+                        ['id' => 5],
+                    ],
+                    'images' => [
+                        ['link' => 'https://via.placeholder.com/300/F00/fff.png'],
+                        ['link' => 'https://via.placeholder.com/300/09f/000.png'],
+                    ],
+                    'mainDetail' => [
+                        'number' => 'FINDOLOGIC1',
+                        'active' => true,
+                        'inStock' => 16,
+                        'prices' => [
+                            [
+                                'customerGroupKey' => 'EK',
+                                'price' => 99.34,
+                            ],
+                        ]
+                    ],
+                ]
+            ],
+            'Article with supplier name space' => [
+                [
+                    'name' => 'FindologicArticle 2',
+                    'active' => true,
+                    'tax' => 19,
+                    'supplier' => " ",
+                    'categories' => [
+                        ['id' => 3],
+                        ['id' => 5],
+                    ],
+                    'images' => [
+                        ['link' => 'https://via.placeholder.com/300/F00/fff.png'],
+                        ['link' => 'https://via.placeholder.com/300/09f/000.png'],
+                    ],
+                    'mainDetail' => [
+                        'number' => 'FINDOLOGIC2',
+                        'active' => true,
+                        'inStock' => 16,
+                        'prices' => [
+                            [
+                                'customerGroupKey' => 'EK',
+                                'price' => 99.34,
+                            ],
+                        ]
+                    ],
+                ]
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider articleSupplierProvider
+     *
+     * @param array $articleConfiguration The article configuration with the corresponding supplier.
+     */
+    public function testEmptySuppliersAreSkipped($articleConfiguration)
+    {
+        try {
+            $baseCategory = new Category();
+            $baseCategory->setId(100);
+
+            $articleFromConfiguration = $this->createTestProduct($articleConfiguration);
+
+            $findologicArticle = $this->articleFactory->create(
+                $articleFromConfiguration,
+                'ABCD0815',
+                [],
+                [],
+                $baseCategory
+            );
+            $this->assertEquals(get_class($findologicArticle), FindologicArticleModel::class);
+        } catch (\Exception $exception) {
+            $this->fail('Exception may not be thrown here!');
```

Comment:
There are two issues with this try/catch:
* `TestCase::assertEquals` throws an exception when it fails. So you will always get `Exception may not be thrown here!`, even if `$this->articleFactory->create` didn't actually throw one.
* The try/catch logic can be removed completely since an unexpected exception causes the test to fail anyway.

Reply:
Done!